scripts/test/delete-cluster.sh

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env bash
+
+
+set -eo pipefail
+
+USAGE=$(cat << 'EOM'
+Usage: delete-cluster.sh -c [cluster-config]
+Deletes an EKS Cluster along with the Nodegroups. Takes in
+the cluster config as an optional argument. If no arugment is
+provided then it will use the cluster config in the default
+path i.e /build/eks-cluster.yaml
+ Optional:
+ -c   eksctl Cluster config path. Defaults to build/eks-cluster.yaml
+ -k   Creation of kOps Cluster. No flag defaults to false.
+ -t   Type of cluster. Defaults to eksctl cluster. Can be set to "kops" or "eksctl"
+EOM
+)
+
+TEST_DIR=$(cd "$(dirname "$0")" || exit 1; pwd)
+CLUSTER_CONFIG_DIR="$TEST_DIR/build"
+CLUSTER_CONFIG_PATH="$CLUSTER_CONFIG_DIR/eks-cluster.yaml"
+
+source "$PWD/scripts/lib/common.sh"
+
+
+check_is_installed eksctl
+
+while getopts "c:t:" o; do
+  case "${o}" in
+    c) # eksctl cluster config file path
+      CLUSTER_CONFIG_PATH=${OPTARG}
+      ;;
+    t) # Type of cluster to be deleted
+      CLUSTER_TYPE=${OPTARG}
+      ;;
+    *)
+      echoerr "${USAGE}"
+      exit 1
+      ;;
+  esac
+done
+
+shift $((OPTIND-1))
+
+function delete_eksctl_cluster {
+    eksctl delete cluster -f "$CLUSTER_CONFIG_PATH" --wait
```

Comment:
Is there a possibility to leak resources like ENI, LB that gets created as part of the test which run on this cluster. ? Do we need to check for them here to clean up or track ?

Reply:
Yes, also VPC or SGs won't be deleted because of dependency.

Reply:
I can check if error is returned on ```eksctl delete``` and  echo "unable to delete cluster ...". Does that sound good ? For other components apart from cluster that need to be cleaned up, the code for those will be in the same modules where they are added. For instance anything applied onto cluster  in another script will have a corresponding cleanup in same script.